src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -73,13 +84,17 @@ public static Summary RunSource(string source, IConfig config = null)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static Summary RunWithDirtyAssemblyResolveHelper(Type type, IConfig config)
-            => BenchmarkRunnerClean.Run(new[] { BenchmarkConverter.TypeToBenchmarks(type, config) }).Single();
+        private static Summary RunWithDirtyAssemblyResolveHelper(Type type, IConfig config, string[] args)
+            => RunWithDirtyAssemblyResolveHelper(new[] { type }, Array.Empty<Assembly>(), config, args);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Summary RunWithDirtyAssemblyResolveHelper(Assembly assembly, IConfig config, string[] args)
+             => RunWithDirtyAssemblyResolveHelper(assembly.GetRunnableBenchmarks().ToArray(), new[] { assembly }, config, args);
```

Comment:
`assembly.GetRunnableBenchmarks().ToArray()` is not needed, `TypeFilter.GetTypesWithRunnableBenchmarks` will take care of it

```suggestion
             => RunWithDirtyAssemblyResolveHelper(Array.Empty<Type>(), new[] { assembly }, config, args);
```